src/GraphQL.Client/Websocket/GraphQLHttpWebSocket.cs

```diff
@@ -407,8 +407,22 @@ public Task InitializeWebSocket()
 #else
                 _clientWebSocket = new ClientWebSocket();
                 _clientWebSocket.Options.AddSubProtocol("graphql-ws");
-                _clientWebSocket.Options.ClientCertificates = ((HttpClientHandler)Options.HttpMessageHandler).ClientCertificates;
-                _clientWebSocket.Options.UseDefaultCredentials = ((HttpClientHandler)Options.HttpMessageHandler).UseDefaultCredentials;
+                try
+                {
+                    _clientWebSocket.Options.ClientCertificates = ((HttpClientHandler)Options.HttpMessageHandler).ClientCertificates;
+                }
+                catch (PlatformNotSupportedException)
+                {
+                    Debug.WriteLine("unable to set Options.ClientCertificates property; platform does not support it");
```

Comment:
It is noop for release build.

Reply:
I know. I figured it was better than just a comment.

Reply:
Probably `Console.WriteLine` (stdout) will be even better.

Reply:
That seems worse as it would interfere with console apps. No production code should ever have any `Console.WriteLine()` statements in it.

Reply:
Well that’s absolutely true for desktop apps. But we should only get the PlatformNotSupported exception in the web where it’s appropriate to display informational logs via the “console“...

Reply:
So let me see if I understand the desired behavior.

**Scenario 1:** The developer does not set these properties, but the code always logs anyway to the web console that it was unable to copy them to `ClientWebSocket` instance.

**Scenario 2:** The developer does attempt to set these properties on the [System.Net.Http.HttpClientHandler](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclienthandler?view=netcore-3.1) instance via `GraphQLHttpClient.Options.HttpMessageHandler` and immediately gets a `PlatformNotSupportedException` error, thus never reaching the code in the try-catch statement and therefore never seeing the web console log message in the first place.

Did I miss something? Because I fail to see when it is useful to log to the web console.

Reply:
I see. Agreed there’s no need for console logs.

Reply:
😄